src/decoder/zlib.rs

```diff
@@ -1,27 +1,22 @@
-use super::{stream::FormatErrorInner, DecodingError, CHUNK_BUFFER_SIZE};
+use super::{stream::FormatErrorInner, DecodingError};
 
 use fdeflate::Decompressor;
 
+pub struct UnfilterBuf<'data> {
```

Comment:
This struct should probably either have its fields marked public or have some public methods so that user code can directly work with `StreamingDecoder::update`.

(Or we should decide that `StreamingDecoder` will no longer be public, but that's a bigger change.)

Reply:
We can probably add a constructor from `&mut Vec<u8>` in some capacity. I concur that would be incredibly helpful for acceptance of this change. Do you have a preference whether we should have a constructor from an intermediary owning struct `(Vec<u8>, usize, usize)` or just one for this? I'd rather not make it fully public in case we have more opportunity for data layout; then this may become an enum internally.

Reply:
I'd be fine with either constructor